CRM/Extension/Mapper.php

```diff
@@ -304,6 +304,7 @@ public function getActiveModuleFiles($fresh = FALSE) {
         FROM civicrm_extension
         WHERE is_active = 1
         AND type = "module"
+        ORDER BY weight ASC
```

Comment:
I agree it's usually preferrable to set per-listener Symfony `$priority` (ie you don't get boxed-in when diff listeners need different priorities).

But... I do think the PR is right to hint that something is off wrt the `civicrm_extension` ordering. The problem is that the default order is *undefined*... so the de-facto order depends on internal+arbitrary DBMS details. You can get two different sites with the same extensions - but they fire in different orders. (*IIRC, it's common for happenstance to loosely mean "first-installed to last-installed", which isn't crazy - but it's not guaranteed, and any specific ordering is hard to reproduce.*)

From that POV, adding a `weight` with default `weight=0` still leaves the ordering undefined. (There's no way to differentiate a bunch of `weight=0`s.) Alternatively, sorting on `full_name` or `name` would be simple + reproducible.

```sql
ORDER BY full_name
```

But I imagine that ordering looks rather different from the  *de-facto* orderings.

Sorting on the dependency-graph might be a little smoother for an install-base where the de-facto orderings most closely approximate "first-installed to last-installed". Something like this, perhaps:

```
cv ev '$m=CRM_Extension_System::singleton()->getManager(); $k=array_keys($m->getStatuses()); sort($k); return array_flip($m->findInstallRequirements($k));'
```

(*Of course, ou'd want to store that ordering somewhere... like in `civicrm_extension.weight` or maybe just a cache...*)